File: rex/lib/ScanDrv.cxx
/* $Id: ScanDrv.cxx,v 1.9 2000/05/25 08:58:02 grosch rel $ */

# include <stdio.h>
# include "Position.h"
$@ # include "@.h"

# define WordSize 2048

int main (void)
{
   int Token, Count = 0;
$@    @ Scanner;

   do {
      Token = Scanner.GetToken ();
      Count ++;
# ifdef Debug
      char Word [WordSize];
      if (Token != EofToken) {
	 if (Scanner.TokenLength < WordSize) {
	    (void) Scanner.GetWord (Word);
	 } else {
	    strncpy (Word, Scanner.TokenPtr, WordSize - 1);
	    Word [WordSize - 1] = '\0';
	 }
      } else {
	 Word [0] = '\0';
      }
      WritePosition (stdout, Scanner.Attribute.Position);
      (void) printf ("%5d %s\n", Token, Word);
# endif
   } while (Token != EofToken);
   (void) printf ("%d\n", Count);
   return 0;
}

File: rex/lib/ScanDrv.java
/* $Id: ScanDrv.java,v 1.4 2000/09/12 17:30:00 grosch rel $ */

$# package declaration
import java.io.*;

$V define xxMaxCharacter

/**
 * Simple class for driving a generated scanner.
 * <P>
 */

$@ class @Drv {

    public static void main (String argv []) throws java.io.IOException {
$@ 	@ scanner = new @ ();
	int token;
	boolean debug = false;
	String filename = null;
	int count = 0;

	for (int i = 0; i < argv.length; i ++) {
	    if (argv [i].equals ("-D")) debug = true;
	    else filename = argv [i];
	}

	if (filename != null)
# if xxMaxCharacter < 256
	    scanner.beginFile (new FileInputStream (filename));
# else
	    scanner.beginFile (new FileReader (filename));
# endif

	do {
	    token = scanner.getToken ();
	    count ++;
	    if (debug) {
		String word = scanner.getWord ();
		System.err.println
		    (scanner.attribute.position () + " " + token + " " + word);
	    }
$@ 	} while (token != @.eofToken);
	scanner.finalize ();
	System.out.println (count);
    }
}
